### PR TITLE
[search] Added a timer and some additional information to the output …

### PR DESCRIPTION
**Pull request by mpimenov**
…of the quality tests.

### CODE REVIEW

File: search/search_quality_tests/search_quality_tests.cpp [hunk 1]
```diff
@@ -69,9 +77,14 @@ void ReadStringsFromFile(string const & path, vector<string> & result)
 // If n == 1, prints the query and the top result separated by a tab.
 // Otherwise, prints the query on a separate line
 // and then prints n top results on n lines starting with tabs.
-void PrintTopResults(string const & query, vector<search::Result> const & results, size_t n)
+void PrintTopResults(string const & query, vector<search::Result> const & results, size_t n,
+                     double elapsedSeconds)
 {
   cout << query;
+  char timeBuf[100];
+  sprintf(timeBuf, "\t[%.3fs]", elapsedSeconds);
```
Comment (ygorshenin) [inline search/search_quality_tests/search_quality_tests.cpp]: snprintf?

Comment (ygorshenin) [inline search/search_quality_tests/search_quality_tests.cpp]: or ostringstream with setprecision?

File: search/search_quality_tests/search_quality_tests.cpp [hunk 2]
```diff
@@ -83,9 +96,24 @@ void PrintTopResults(string const & query, vector<search::Result> const & result
...
       cout << kEmptyResult;
   }
+  if (n == 1)
+    cout << timeBuf;
   cout << endl;
 }
 
+uint64_t ReadVersionFromHeader(platform::LocalCountryFile const & mwm)
+{
+  if (mwm.GetCountryName() == "World" || mwm.GetCountryName() == "WorldCoasts")
```
Comment (ygorshenin) [inline search/search_quality_tests/search_quality_tests.cpp]: WORLD_FILE_NAME
WORLD_COASTS_FILE_NAME
from defines.hpp

File: search/search_quality_tests/search_quality_tests.cpp [hunk 3]
```diff
@@ -122,11 +150,18 @@ int main(int argc, char * argv[])
...
+  for (auto & mwm : mwms)
   {
-    LOG(LINFO, ("Registering map:", mwm.GetCountryName()));
+    mwm.SyncWithDisk();
+    cout << mwm.GetCountryName() << " " << ReadVersionFromHeader(mwm) << endl;
     engine.RegisterMap(mwm);
   }
+  cout << endl;
+
+  // auto const & viewport = kDefaultViewport;
```
Comment (ygorshenin) [inline search/search_quality_tests/search_quality_tests.cpp]: How about to add a command line flag for default viewports, like Moscow, London, Zurich, Munich?

File: search/search_quality_tests/search_quality_tests.cpp (hunk 1 above)
Comment (mpimenov) [inline search/search_quality_tests/search_quality_tests.cpp]: done

File: search/search_quality_tests/search_quality_tests.cpp (hunk 2 above)
Comment (mpimenov) [inline search/search_quality_tests/search_quality_tests.cpp]: done

File: search/search_quality_tests/search_quality_tests.cpp (hunk 3 above)
Comment (mpimenov) [inline search/search_quality_tests/search_quality_tests.cpp]: done

**Comment by mpimenov** [comment]
PTAL

File: search/search_quality_tests/search_quality_tests.cpp [hunk 4]
```diff
@@ -122,11 +157,29 @@ int main(int argc, char * argv[])
...
+    string name = FLAGS_viewport;
+    auto it = kViewports.find(name);
+    if (it == kViewports.end())
+    {
+      name = "default";
+      it = kViewports.find(name);
+    }
+    CHECK(it != kViewports.end(), ());
+    viewport = it->second;
+    cout << "Viewport used in all search invocations: " << name << " " << DebugPrint(viewport)
```
Comment (ygorshenin) [inline search/search_quality_tests/search_quality_tests.cpp]: s/\n\n/endl endl/g

**Comment by mpimenov** [comment]
Добавил вывод среднего.

**Comment by mpimenov** [comment]
PTAL

File: search/search_quality_tests/search_quality_tests.cpp (hunk 4 above)
Comment (mpimenov) [inline search/search_quality_tests/search_quality_tests.cpp]: done

File: search/search_quality_tests/search_quality_tests.cpp [hunk 5]
```diff
@@ -134,15 +191,36 @@ int main(int argc, char * argv[])
...
+    averageTime += responseTimes[i];
+    maxTime = max(maxTime, responseTimes[i]);
   }
+  if (responseTimes.size() > 0)
+    averageTime /= static_cast<double>(responseTimes.size());
+  double stdDevTime = 0;
+  for (auto const t : responseTimes)
+    stdDevTime += math::sqr(t - averageTime);
+  if (responseTimes.size() > 0)
+    stdDevTime /= static_cast<double>(responseTimes.size());
```
Comment (ygorshenin) [inline search/search_quality_tests/search_quality_tests.cpp]:
```c++
responseTimes.size() - 1;
```
For non-skewed sd.

File: search/search_quality_tests/search_quality_tests.cpp [hunk 6]
```diff
@@ -134,15 +191,36 @@ int main(int argc, char * argv[])
...
+  stdDevTime = sqrt(stdDevTime);
+
+  cout << fixed << setprecision(3);
+  cout << endl;
+  cout << "Maximum response time: " << maxTime << "s" << endl;
+  cout << "Average response time: " << averageTime << "s"
+       << " (std. dev. " << stdDevTime << "s)" << endl;
```
Comment (ygorshenin) [inline search/search_quality_tests/search_quality_tests.cpp]: It's not a stdDev, it's a variance. To compute stdDev, please, apply an sqrt().

**Comment by mpimenov** [comment]
PTAL

**Comment by ygorshenin** [comment]
LGTM

**Pull request merged**